### PR TITLE
add prompt for seed when restoring wallet

### PR DESCRIPTION
**Pull request by mj10021**
Fixes issue #2990 by allowing a seed to be entered to stdin when restoring a wallet.  I'm not sure if I got the type coercion/checking exactly right here, so would appreciate any feedback/help.  Thanks!

### CODE REVIEW

File: src/subcommand/wallet/restore.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
 use super::*;
+use std::io;
```
Comment (raphjaph) [inline src/subcommand/wallet/restore.rs:2]: combine these two
```rust
use {
  super::*,
  std::io,
};
```

File: src/subcommand/wallet/restore.rs [hunk 2]
```diff
@@ -14,10 +15,20 @@ pub(crate) struct Restore {
 
 impl Restore {
   pub(crate) fn run(self, wallet_name: String, options: Options) -> SubcommandResult {
+    let seed = {
+      if self.mnemonic.is_none() {
+        let mut input = String::new();
+        println!("Please input your seed below:");
+        io::stdin().read_line(&mut input).expect("failed to read mnemonic");
+        let input = input.into_bytes();
+        assert_eq!(input.len(), 64);
```
Comment (raphjaph) [inline src/subcommand/wallet/restore.rs:24]: not sure if we need this the `Mnemonic` might already perform these checks.

**Review by raphjaph — CHANGES_REQUESTED** [review]
This is good start! Definitely add a an integration test somewhere in `tests/wallet`. If you need help with that let me know.

**Comment by raphjaph** [comment]
@mj10021 Have a look at #3048. I read the descriptor from stdin and also have some tests in there that you can use as a template.

**Comment by mj10021** [comment]
thanks :) i'm away this weekend but should have done early next week

**Comment by mj10021** [comment]
hey @raphjaph , just to clarify, is the stdin meant to accept a descriptor or a mnemonic?

**Comment by mj10021** [comment]
> hey @raphjaph , just to clarify, is the stdin meant to accept a descriptor or a mnemonic?

Oh I see, the --descriptor flag lets you enter a descriptor through stdin without the mnemonic

**Comment by mj10021** [comment]
Ok, sorry, I actually do need some clarification. I see that now there is an option to restore from stdin and tests have already been made for that. Is this PR now redundant? Or is the option to load a mnemonic from stdin in addition to the descriptor flag also wanted? Thanks!

**Comment by raphjaph** [comment]
> Ok, sorry, I actually do need some clarification. I see that now there is an option to restore from stdin and tests have already been made for that. Is this PR now redundant? Or is the option to load a mnemonic from stdin in addition to the descriptor flag also wanted? Thanks!

Yes we want to load the mnemonic from stdin only as well. And def also add a separate test for that

**Comment by mj10021** [comment]
> > Ok, sorry, I actually do need some clarification. I see that now there is an option to restore from stdin and tests have already been made for that. Is this PR now redundant? Or is the option to load a mnemonic from stdin in addition to the descriptor flag also wanted? Thanks!
> 
> Yes we want to load the mnemonic from stdin only as well. And def also add a separate test for that. Let me know when this is ready for review.

Ok got it thanks.  I'm not able to run the tests atm but I think this should be right?

**Comment by raphjaph** [comment]
Tests aren't passing yet. Install `just` and run `just ci` to run all our tests.

**Comment by mj10021** [comment]
> Tests aren't passing yet. Install `just` and run `just ci` to run all our tests.

Got it, thanks.  Should be good now.

**Review by raphjaph — APPROVED** [review]
LGTM

**Pull request merged**